trucks.js

```diff
@@ -502,3 +502,46 @@ var foodTrucks = [
 		Twitter: 'https://twitter.com/fticecream'
 	}
 ];
+
+var _ = require('underscore');
+function getTrucks() {
+	return foodTrucks;
+}
+
+function getTruck(name) {
+	return _.find(foodTrucks, function(truck) {
+		return truck.name === name;
+	});
+
+}
+
+function getFoodTypes(){
+	var masterList = _.map(foodTrucks, function(truck){
+		return truck.type;
+	});
+	var flattenedArrays = _.flatten(masterList);
+	var uniqList = _.uniq(flattenedArrays);
+	return uniqList;
+}
+
+
+function addTruck(truck) {
+	if (truck){
+	foodTrucks.push(truck);
+}
+}
+
+function removeTruck(name) {
+	var truckIndex = _.findIndex(foodTrucks,function (truck) {
+		return truck.name === name;
+	});
+	foodTrucks.splice(truckIndex, 1);
+}
+
+module.exports = {
+	getTrucks: getTrucks,
+	getTruck: getTrucks,
```

Comment:
I was completely baffled by the behavior of your '/trucks/:name' route until I finally looked at this, which should be:

`getTruck: getTruck` instead of `getTruck: getTrucks`

the 's' always gets me